### PR TITLE
Enhance build script for Macos and Ubuntu

### PR DESCRIPTION
**Pull request by leoluan2009**
Enhance ubuntu build script for presto_cpp project, presto_cpp will add build script soon like script for Macos system which will use some function added in this PR.
Fix a undefined variable error in setup-macos.sh when this file was used by presto_cpp build script

### CODE REVIEW

File: scripts/setup-macos.sh [hunk 1]
```diff
@@ -42,7 +42,7 @@ function run_and_time {
 function prompt {
   (
     while true; do
-      local input="${PROMPT_ALWAYS_RESPOND}"
+      local input="${PROMPT_ALWAYS_RESPOND:-}"
```
Comment (pedroerp) [inline scripts/setup-macos.sh:45]: what does this do?

File: scripts/setup-ubuntu.sh [hunk 2]
```diff
@@ -14,17 +14,14 @@
...
 export COMPILER_FLAGS="-mavx2 -mfma -mavx -mf16c -masm=intel -mlzcnt"
-BUILD_DIR=_build
-
-set -eufx -o pipefail
-
-# Enter build directory.
-mkdir -p "${BUILD_DIR}"
-cd "${BUILD_DIR}"
+FB_OS_VERSION=v2021.05.10.00
+NPROC=$(getconf _NPROCESSORS_ONLN)
```
Comment (pedroerp) [inline scripts/setup-ubuntu.sh:23]: where are you planning to use this variable?

File: scripts/setup-ubuntu.sh [hunk 3]
```diff
@@ -52,31 +49,86 @@ sudo apt install -y \
   liblzo2-dev \
   protobuf-compiler
 
-function install_folly {
-  local NAME="folly"
+function run_and_time {
```
Comment (pedroerp) [inline scripts/setup-ubuntu.sh:52]: maybe not on this PR, but it would be nice to move this common code to some sort of library at some point.

**Review by pedroerp — APPROVED** [review]
Thanks for the contribution, this looks good!

File: scripts/setup-macos.sh (hunk 1 above)
Comment (leoluan2009) [reply scripts/setup-macos.sh:45]: fix undefined variable error when this file was used by presto_cpp at https://github.com/facebookexternal/presto_cpp/blob/c96a88c89489574dba8b791c50a24b4a52a559ae/scripts/setup-macos.sh#L17, and the file has a -u flag. The error will cause rebuilding dependencies every time

File: scripts/setup-ubuntu.sh (hunk 2 above)
Comment (leoluan2009) [reply scripts/setup-ubuntu.sh:23]: Will be used in presto_cpp project

**Comment by leoluan2009** [comment]
@pedroerp  can you help to check the failed of Facebook Internal - Linter, I don not have Permission to see the error  details.

**Pull request closed without merging**